query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoFilterHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.hypertrace.core.query.service.trino;
+
+import static org.hypertrace.core.query.service.QueryRequestUtil.getLogicalColumnName;
+import static org.hypertrace.core.query.service.api.Expression.ValueCase.LITERAL;
+
+import java.util.List;
+import java.util.Optional;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.api.ValueType;
+
+public class TrinoFilterHandler {
+  private static final String IS_TRINO_ATTRIBUTE = "Event.isTrino";
+
+  public boolean containsAttributeFilter(QueryRequest request) {
+    return request.hasFilter() && containsAttributeFilter(request.getFilter());
+  }
+
+  public Filter skipTrinoAttributeFilter(Filter filter) {
+    return skipAttributeFilterIfPresent(filter);
+  }
+
+  private boolean containsAttributeFilter(Filter filter) {
+    if (filter.getChildFilterCount() > 0) {
+      for (Filter childFilter : filter.getChildFilterList()) {
+        if (containsAttributeFilter(childFilter)) {
+          return true;
+        }
+      }
+    }
+    return isTrinoAttributeFilter(filter);
+  }
+
+  private Filter skipAttributeFilterIfPresent(Filter filter) {
+    if (filter.getChildFilterCount() > 0) {
```

Comment:
Do we have to test if filter is default instance (or null)?

Reply:
Should not be needed now after adding request.hasFilter() check